pybamm/expression_tree/averages.py

```diff
@@ -0,0 +1,366 @@
+#
+# Classes and methods for averaging
+#
+import pybamm
+
+
+class _BaseAverage(pybamm.Integral):
+    """
+    Base class for a symbol representing an average
+
+    Parameters
+    -----------
+    child : :class:`pybamm.Symbol`
+        The child node
+    """
+
+    def __init__(self, child, name, integration_variable):
+        super().__init__(child, integration_variable)
+        self.name = name
+
+
+class XAverage(_BaseAverage):
+    def __init__(self, child):
+        if child.domain in [
+            ["negative particle"],
+            ["negative particle size"],
+        ]:
+            x = pybamm.standard_spatial_vars.x_n
+        elif child.domain in [
+            ["positive particle"],
+            ["positive particle size"],
+        ]:
+            x = pybamm.standard_spatial_vars.x_p
+        else:
+            x = pybamm.SpatialVariable("x", domain=child.domain)
+        integration_variable = x
+        super().__init__(child, "x-average", integration_variable)
+
+    def _unary_new_copy(self, child):
+        """See :meth:`UnaryOperator._unary_new_copy()`."""
+        return x_average(child)
+
+
+class YZAverage(_BaseAverage):
+    def __init__(self, child):
+        y = pybamm.standard_spatial_vars.y
+        z = pybamm.standard_spatial_vars.z
+        integration_variable = [y, z]
+        super().__init__(child, "yz-average", integration_variable)
+
+    def _unary_new_copy(self, child):
+        """See :meth:`UnaryOperator._unary_new_copy()`."""
+        return yz_average(child)
+
+
+class ZAverage(_BaseAverage):
+    def __init__(self, child):
+        integration_variable = [pybamm.standard_spatial_vars.z]
+        super().__init__(child, "z-average", integration_variable)
+
+    def _unary_new_copy(self, child):
+        """See :meth:`UnaryOperator._unary_new_copy()`."""
+        return z_average(child)
+
+
+class RAverage(_BaseAverage):
+    def __init__(self, child):
+        integration_variable = [pybamm.SpatialVariable("r", child.domain)]
+        super().__init__(child, "r-average", integration_variable)
+
+    def _unary_new_copy(self, child):
+        """See :meth:`UnaryOperator._unary_new_copy()`."""
+        return r_average(child)
+
+
+class SizeAverage(_BaseAverage):
+    def __init__(self, child, f_a_dist):
+        R = pybamm.SpatialVariable(
+            "R",
+            domain=child.domain,
+            auxiliary_domains=child.auxiliary_domains,
+            coord_sys="cartesian",
+        )
+        integration_variable = [R]
+        super().__init__(child, "size-average", integration_variable)
+        self.f_a_dist = f_a_dist
+
+    def _unary_new_copy(self, child):
+        """See :meth:`UnaryOperator._unary_new_copy()`."""
+        return size_average(child, f_a_dist=self.f_a_dist)
+
+
+def x_average(symbol):
+    """
+    convenience function for creating an average in the x-direction.
```

Comment:
```suggestion
    Convenience function for creating an average in the x-direction.
```

Not sure if what I am suggesting goes against the standard format. If not, note that the change also applies to all the other functions below.